tools/go-cli/go-whisk-cli/commands/util.go

```diff
@@ -99,177 +98,75 @@ func parseQualifiedName(name string) (qName qualifiedName, err error) {
-    data := []byte(res)
-
-    whisk.Debug(whisk.DbgInfo, "Parsed parameters: %s\n", res)
-
-    return (*json.RawMessage)(&data), nil
-}
-
-func parseParametersArray(args []string) (*json.RawMessage, error) {
-    var res string
-
-    if len(args) == 0 {
-        whisk.Debug(whisk.DbgInfo, "parseParametersArray: no args\n")
-
-        return nil, nil
-    }
-
-    if len(args)%2 != 0 {
-        whisk.Debug(whisk.DbgError, "Number of arguments (%d) must be an even number; args: %#v\n", len(args), args)
-        err := whisk.MakeWskError(
-            errors.New("key|value arguments must be submitted in comma-separated pairs; keys or values with spaces must be quoted"),
-            whisk.EXITCODE_ERR_GENERAL, whisk.DISPLAY_MSG, whisk.DISPLAY_USAGE )
-        return nil, err
-    }
-
-    for i := 0; i < len(args); i += 2 {
-        res = res + "{\"key\": \"" + args[i] + "\", "
-
-        if ((len(args[i + 1]) > 0) && (isValidJSON(args[i + 1]) || (args[i + 1][0] == '[' && args[i + 1][len(args[i + 1]) - 1] == ']') ||
-        (args[i + 1][0] == '"' && args[i + 1][len(args[i + 1]) - 1] == '"'))) {
-            whisk.Debug(whisk.DbgInfo, "Parameters are valid JSON: %s\n", args[i+1])
-
-            res = res + "\"value\": " + args[i+1] + "}"
-        } else {
-            whisk.Debug(whisk.DbgInfo, "Parameters are not valid JSON: %s\n", args[i+1])
-
-            _, err := strconv.Atoi(args[i + 1])
-
-            if err == nil {
-                res = res + "\"value\": " + args[i + 1] + "}"
-            } else {
-                res = res + "\"value\": \"" + args[i + 1] + "\"}"
-            }
-        }
-
-        if i < len(args) - 3 {
-            res = res + ", "
-        }
+    if keyValueFormat {
+        res = fmt.Sprintf("[%s]", res)
+    } else {
+        res = fmt.Sprintf("{%s}", res)
     }
 
-    res = "[" + res + "]"
-    res = strings.Replace(res, "\n", "\\n", -1)
     data := []byte(res)
 
-    whisk.Debug(whisk.DbgInfo, "Parsed parameters: %s\n", res)
-
-
+    whisk.Debug(whisk.DbgInfo, "getJSONFromArguments: JSON: %s\n", res)
     return (*json.RawMessage)(&data), nil
 }
 
-func parseAnnotations(args []string) (whisk.Annotations, error) {
-    annotations := whisk.Annotations{}
+func getJSONKeyValueFromArguments(key string, value string) (string, string) {
+    key = getEscapedString(key)
 
-    parsedArgs, err := parseGenericArray(args)
-    if err != nil {
-        whisk.Debug(whisk.DbgError, "parseGenericArray(%#v) error: %s\n", args, err)
-        return annotations, err
+    if (len(value) == 0 || !isValidJSON(value)) {
+        whisk.Debug(whisk.DbgInfo,
+            "getJSONKeyValueFromArguments(): Assuming string for input as it is not valid JSON: %s\n", value)
+        value = fmt.Sprintf("\"%s\"", getEscapedString(value))
     }
 
-    annotations = whisk.Annotations(parsedArgs)
+    return key, value
+}
 
-    return annotations, nil
+func getEscapedString(value string) (string) {
```

Comment:
This function is escaping actual whitespace characters, backslashes, and qoutes to have parity with the Python CLI.